hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/action/cluster/strategy/ClusteringExecutionStrategy.java

```diff
@@ -140,7 +142,8 @@ protected FileSlice clusteringOperationToFileSlice(String basePath, ClusteringOp
     String baseInstantTime = baseFile.map(HoodieBaseFile::getCommitTime).orElseGet(() -> logFiles.get(0).getDeltaCommitTime());
     FileSlice fileSlice = new FileSlice(partitionPath, baseInstantTime, clusteringOperation.getFileId());
     baseFile.ifPresent(fileSlice::setBaseFile);
-    logFiles.forEach(fileSlice::addLogFile);
+    CompletionTimeQueryView completionTimeQueryView = metaClient.getTimelineLayout().getTimelineFactory().createCompletionTimeQueryView(metaClient);
```

Comment:
It does not make any sense to instantiate the `CompletionTimeQueryView` in file slice level, -1.